client/my-sites/upgrades/components/domain-warnings/index.jsx

```diff
@@ -68,6 +86,58 @@ export default React.createClass( {
 		return ( this.props.domains || [ this.props.domain ] ).filter( domain => domain.currentUserCanManage );
 	},
 
+	wrongNSMappedDomains() {
+		debug( 'Rendering wrongNSMappedDomains' );
+		const wrongMappedDomains = this.getDomains().filter( domain =>
+			domain.type === domainTypes.MAPPED && ! domain.pointsToWpcom );
+
+		debug( 'Domains:', this.getDomains() );
+		debug( 'NS error domains:', wrongMappedDomains );
+		let learnMoreUrl,
+			text,
+			offendingList = null;
+
+		if ( wrongMappedDomains.length === 0 ) {
+			return null;
+		}
+
+		if ( wrongMappedDomains.length === 1 ) {
+			const domain = wrongMappedDomains[ 0 ];
+			if ( isSubdomain( domain.name ) ) {
+				text = this.translate( '%(domainName)s\'s CNAME records should be configured', {
+					args: { domainName: domain.name },
+					context: 'Notice for mapped subdomain that has CNAME records need to set up'
+				} );
+				learnMoreUrl = support.MAP_SUBDOMAIN;
+			} else {
+				text = this.translate( '%(domainName)s\'s name server records should be configured', {
+					args: { domainName: domain.name },
+					context: 'Notice for mapped domain notice with NS records pointing to somewhere else'
+				} );
+				learnMoreUrl = support.DOMAIN_HELPER_PREFIX + domain.name;
+			}
+		} else {
+			offendingList = <ul>{ wrongMappedDomains.map( domain => <li key={ domain.name }>{ domain.name }</li> ) }</ul>;
+			if ( every( map( wrongMappedDomains, 'name' ), isSubdomain ) ) {
+				text = this.translate( 'Some of your domains\' CNAME records should be configured', {
+					context: 'Notice for mapped subdomain that has CNAME records need to set up'
+				} );
+				learnMoreUrl = support.MAP_SUBDOMAIN;
+			} else {
+				text = this.translate( 'Some of your domains\' name server records should be configured', {
+					context: 'Mapped domain notice with NS records pointing to somewhere else'
+				} );
+				learnMoreUrl = support.MAP_EXISTING_DOMAIN_UPDATE_DNS;
+			}
+		}
+		return <Notice
+			isCompact={ this.props.isCompact }
+			status="is-warning"
+			className="domain-warnings-notice"
+			showDismiss={ false }
+			key="wrong-ns-mapped-domain">{ text } <a href={ learnMoreUrl }>{ this.translate( 'Learn more.' ) }</a>{ offendingList }</Notice>;
```

Comment:
Also, can we _please_ move the dot in the sentence to the left a bit? Currently it looks off to me:
<img width="958" alt="screen shot 2016-08-01 at 01 19 12" src="https://cloud.githubusercontent.com/assets/3392497/17280033/3254ab1c-5786-11e6-9446-e7f2c5572eb1.png">

It's especially stands out in the compact version:
<img width="285" alt="screen shot 2016-08-01 at 01 20 13" src="https://cloud.githubusercontent.com/assets/3392497/17280039/4f605f62-5786-11e6-8e60-a967ac718ce9.png">

This seems more consistent with our other warning notices:
<img width="952" alt="screen shot 2016-08-01 at 01 19 53" src="https://cloud.githubusercontent.com/assets/3392497/17280037/40ecaa62-5786-11e6-8d2f-8b7bb2e9e15c.png">

Sorry for pointing out such minute detail, but sometimes such details are what make the UI seem more polished and professional. Also, you know I like to nitpick 😝